import_data/osm_update.py

```diff
@@ -0,0 +1,240 @@
+        log("no expired tiles")
+        return True
+
+    log("file with tile to regenerate = {}".format(entries))
+
+    args = ["invoke"]
+    if settings["invoke_option"] != "":
+        args.append(settings["invoke_option"])
+    args.extend([
+        "generate-expired-tiles",
+        "--tiles-layer", json_data["tiles_layer_name"],
+        "--from-zoom", settings["from_zoom"],
+        "--before-zoom", settings["before_zoom"],
+        "--expired-tiles", entries
+    ])
+    if exec_command(args) != 0:
+        log_error("Failed to run command `{}`".format(args))
+        return False
+    return True
+
+
+def parse_args(settings):
+    pos = 1
+    while pos < len(sys.argv):
+        arg = sys.argv[pos]
+        if arg in ("-c", "--config"):
+            pos += 1
+            if pos >= len(sys.argv):
+                log_error("Missing argument after `{}` option".format(arg))
+            settings["imposm_config_dir"] = sys.argv[pos]
+        elif arg in ("-i", "--input"):
+            pos += 1
+            if pos >= len(sys.argv):
+                log_error("Missing argument after `{}` option".format(arg))
+            settings["change_file"] = sys.argv[pos]
+        elif arg in ("-h", "--help"):
+            print_usage()
+        elif arg == "--":
+            break
+        else:
+            log_error("Unknown option `{}`".format(arg))
+            return False
+        pos += 1
+    return True
+
+
+def run(settings):
+    settings = settings.copy() # to not modify original
+    invoke_config_file = os.environ.get("INVOKE_CONFIG_FILE", "")
+    # Settings
+    settings["start"] = datetime.now()
+    settings["osm_update_working_dir"] = os.environ.get("OSM_UPDATE_WORKING_DIR", "/data/osmosis")
+    settings["osmosis_working_dir"] = os.environ.get("OSMOSIS_WORKING_DIR", "/data/osmosis")
+    settings["exec_time"] = get_time_now()
+    settings["invoke_config_file"] = invoke_config_file
+    settings["invoke_option"] = ""
+    if invoke_config_file != "":
+        settings["invoke_option"] = "-f {}".format(invoke_config_file)
+    # imposm
+    if settings.get("imposm_config_dir", "") == "":
+        # default value, can be set with the --config option
+        settings["imposm_config_dir"] = "/etc/imposm"
+    settings["imposm_data_dir"] = os.environ.get("IMPOSM_DATA_DIR", "/data/imposm")
+    # base tiles
+    settings["base_imposm_config_filename"] = "config_base.json"
+    # poi tiles
+    settings["poi_imposm_config_filename"] = "config_poi.json"
+    # tilerator
+    settings["from_zoom"] = 11
+    settings["before_zoom"] = 15 # exclusive
+
+    log("new osm_update process started")
+    log("working into directory: {}".format(settings["osm_update_working_dir"]))
+
+    if os.environ.get("PGPASS") is None or not path.isfile(os.environ["PGPASS"]):
+        log_error("PostgreSQL user `{}` password file `{}` not found!".format(
+            os.environ["PGUSER"],
+            os.environ.get("PGPASS", "")))
+        return False
```

Comment:
These variables are not used and these checks are probably obsolete.

Reply:
Removed.